sirepo/package_data/template/omega/parameters.py.jinja

```diff
@@ -80,31 +110,58 @@ def run_genesis(run_dir, genesis_id, prev_sim=None):
     data = read_sim('genesis', genesis_id)
     if prev_sim:
         assert prev_sim.outfile_path
-        filename = file_name_from_sim_name('genesis', f'{_OMEGA_SIM_NAME}-{run_dir.basename}')
-        sw = switchyard.Switchyard()
-        sw.read(f'{prev_sim.outfile_path}', prev_sim.sim_type)
-        lib_file = sim_lib_file('genesis', 'io', 'distfile', filename)
-        sw.write(lib_file, 'genesis')
+        filename = file_name_from_sim_name(
+            "genesis", f"{_OMEGA_SIM_NAME}-{run_dir.basename}"
+        )
+        lib_file = sim_lib_file("genesis", "io", "partfile", filename)
+        if prev_sim.sim_type == "genesis":
+            # center longitudinal
+            with open(prev_sim.outfile_path, 'rb') as f:
+                d = numpy.fromfile(f, dtype=numpy.float64)
+            d = d.reshape((6, int(len(d) / 6)))
+            d[1,:] -= numpy.mean(d[1,:])
+            with open(lib_file, 'wb') as f:
+                d.tofile(f)
+        else:
+            if prev_sim.sim_type == "elegant":
+                p = switchyard.read_elegant(f"{prev_sim.outfile_path}")
+            else:
+                assert prev_sim.sim_type == "opal"
+                p = switchyard.read_opal(f"{prev_sim.outfile_path}")
+            particle_data = numpy.zeros([6, len(p.x)])
+            for i, col in enumerate(['pt', 'ct', 'x', 'y', 'ux', 'uy']):
```

Comment:
maybe store `['pt', 'ct', 'x', 'y', 'ux', 'uy']` in a constant that comes from the template?

Reply:
These items are the mapping from switchyard to genesis, but there isn't a full definition for it outside this file.